doc/user/content/sql/isolation-level.md

```diff
@@ -0,0 +1,96 @@
+---
+title: "Isolation Level"
+description: "The isolation level defines how Materialize isolates the execution of transactions."
+aliases:
+ - /sql/consistency
+menu:
+  main:
+    parent: 'reference'
+    weight: 160
+---
+
+The SQL standard defines four levels of transaction isolation. In order of least strict to most strict they are:
+
+  * Read Uncommitted
+  * Read Committed
+  * Repeatable Read
+  * Serializable
+
+In Materialize, you can request any of these isolation
+levels, but they all behave the same as the Serializable isolation level. In addition to the four levels defined in the
+SQL Standard, Materialize also defines a [Strict Serializable](#strict-serializable) isolation level.
+
+Isolation level is a per session configurable variable. To set the current session’s isolation level you can
+execute `SET TRANSACTION_ISOLATION TO '<isolation-level>';`. The default isolation level is Strict Serializable.
+
+## Serializable
+
+The SQL standard defines the Serializable isolation level as preventing the following three phenomenons:
+
+- **P1 (”Dirty Read”):**
+  > SQL-transaction T1 modifies a row. SQL-transaction T2 then reads that row before T1 performs a
+  COMMIT. If T1 then performs a ROLLBACK, T2 will have read a row that was never committed and that may thus be
+  considered to have never existed.
+
+- **P2 (”Non-repeatable read”):**
+
+  > SQL-transaction T1 reads a row. SQL-transaction T2 then modifies or deletes that row and performs
+  a COMMIT. If T1 then attempts to reread the row, it may receive the modified value or discover that the row has been
+  deleted.
+
+- **P3 (”Phantom”):**
+
+  > SQL-transaction T1 reads the set of rows N that satisfy some <search condition>. SQL-transaction
+  T2 then executes SQL-statements that generate one or more rows that satisfy the <search condition> used by
+  SQL-transaction T1. If SQL-transaction T1 then repeats the initial read with the same <search condition>, it obtains a
+  different collection of rows.
+
+Furthermore, Serializable also guarantees that the result of executing a group of concurrent SQL-transactions produces
+the same effect as some serial execution of those same transactions. A serial execution is one where each
+SQL-transaction executes to completion before the next one begins. There is no guarantee that this serial ordering is
+consistent with the real time ordering of the transactions, in other words transactions are not linearizable under the
+Serializable isolation level. For example if SQL-transaction T1 happens before SQL-transaction T2 in real time, then the
+result may be equivalent to a serial order where T2 was executed first.
+
+In practice non-linearizable orderings are unlikely to happen, but still possible, when querying directly from tables
```

Comment:
This text about unlikeliness seems sketchy to me. I think we have no strong idea about how likely it will be for use cases we don't control, and especially if we end up relaxing `Serializable` from its current behavior to allow even lower latency.

Reply:
Yeah I was a bit iffy on it as well. I will remove that part.

Reply:
Removed